pymc_bart/pgbart.py

```diff
@@ -415,18 +421,24 @@ def grow_tree(
         get_idx_right_child(index_leaf_node),
     )
 
+    if response == "mix":
+        response = "linear" if np.random.random() >= 0.5 else "constant"
+
     for idx in range(2):
         idx_data_point = new_idx_data_points[idx]
-        node_value = draw_leaf_value(
-            sum_trees[:, idx_data_point],
-            m,
-            normal.rvs() * kfactor,
-            shape,
+        node_value, linear_params = draw_leaf_value(
+            y_mu_pred=sum_trees[:, idx_data_point],
+            x_mu=X[idx_data_point, selected_predictor],
```

Comment:
@aloctavodia  I think the problem is here as we are not handling the missing values for the linear model anywhere. A brute force approach adding `np.nan_to_num(X[idx_data_point, selected_predictor])` (see [numpy.nan_to_num](https://numpy.org/doc/stable/reference/generated/numpy.nan_to_num.html)) makes the model run. We could fill in missing values or simply warn the user missing values with `linear` response (and therefore `response = mix`) is not supported? 

I do not see anywhere in the PR in the PyMC repo any type of imputation.

Reply:
Thanks for the tip, I resolved thin in https://github.com/pymc-devs/pymc-bart/pull/79/commits/c24262803a5d443145bf423502b7ec2cfa434c1e